File: data-structures/lists/python/list_add.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0


def display_astronauts(astronauts):
    """Display the values from the given astronaut list"""

    print("\nAstronauts:")
    
    # Repeat for each value in the list
    for name in astronauts:
        print(name)


def main():
    # Declare and initialise the astronaut list
    astronauts = ["Sergey Prokopyev", "Frank Rubio", "Stephen Bowen"]

    # Ask the user to input another astronaut
    name = input("Enter the name of another astronaut: ")
    
    # Add the name to the end of the list
    astronauts.append(name)

    # Output the astronaut list
    display_astronauts(astronauts)


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()

File: data-structures/lists/java/list_add.java
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science
Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.util.ArrayList;
import java.util.Scanner;

class SingleLists {
    // The main method is the entry point for all Java programs
    public static void main(String[] args) {
        // Declare and initialise the astronaut list
        ArrayList<String> astronauts = new ArrayList<String>();
        astronauts.add("Sergey Prokopyev");
        astronauts.add("Frank Rubio");
        astronauts.add("Stephen Bowen");

        // Use Scanner for reading user input from the console
        Scanner scanner = new Scanner(System.in);

        // Ask the user to input another astronaut
        System.out.println("Enter the name of another astronaut:");
        String name = scanner.nextLine();

        // Add the name to the end of the list
        astronauts.add(name);

        // Output the astronaut list
        displayAstronauts(astronauts);
    }

    // Display the values from the given astronaut list
    public static void displayAstronauts(ArrayList<String> astronauts) {
        System.out.println("\nAstronauts:");

        // Repeat for each value in the list
        for (String name : astronauts) {
            System.out.println(name);
        }
    }
}